ledger/acctonline.go

```diff
@@ -454,6 +475,24 @@ func (ao *onlineAccounts) postCommit(ctx context.Context, dcc *deferredCommitCon
 func (ao *onlineAccounts) postCommitUnlocked(ctx context.Context, dcc *deferredCommitContext) {
 }
 
+// OnlineTotals return the total online balance for the given round. It would replace the existing accountupdates.Total(rnd).
+func (ao *onlineAccounts) OnlineTotals(rnd basics.Round) (basics.MicroAlgos, error) {
+	ao.accountsMu.RLock()
+	defer ao.accountsMu.RUnlock()
+	return ao.onlineTotalsImpl(rnd)
+}
+
+// onlineTotalsImpl returns the online totals of all accounts at the end of round rnd.
+func (ao *onlineAccounts) onlineTotalsImpl(rnd basics.Round) (basics.MicroAlgos, error) {
+	offset, err := ao.roundOffset(rnd)
```

Comment:
it works only while we have len(ao.roundOffset) == len(ao.deltas) == MaxBalLookback, in other words, while we store all deltas in memory.
I agree with `lookupOnline` also uses this fact and will need to be changed.